tests/CloneTest.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+namespace ipl\Tests\Html;
+
+use Closure;
+use ipl\Html\Attribute;
+use ipl\Html\Attributes;
+use ipl\Tests\Html\lib\CloningDummyElement;
+use ReflectionException;
+use ReflectionFunction;
+use ReflectionProperty;
+
+class CloneTest extends TestCase
+{
+    public function testHtmlOutput(): void
+    {
+        $original = new CloningDummyElement();
+        $original->getAttributes()->set('class', 'original_class');
+
+        $firstClone = clone $original;
+        $firstClone->getAttributes()->set('class', 'first_clone_class');
+
+        $secondClone = clone $firstClone;
+        $secondClone->getAttributes()->set('class', 'second_clone_class');
+
+        $originalHtml = '<p '
+            . 'class="original_class" '
+            . 'test-instance-scope-noop-inline="inline" '
+            . 'test-instance-noop-attribute="static_callback" '
+            . 'test-closure-static-scope-noop="static_callback" '
+            . 'test-closure-instance-scope-noop="static_callback"'
+            . '></p>';
+
+        $firstCloneHtml = '<p '
+            . 'class="first_clone_class" '
+            . 'test-instance-scope-noop-inline="inline" '
+            . 'test-instance-noop-attribute="static_callback" '
+            . 'test-closure-static-scope-noop="static_callback" '
+            . 'test-closure-instance-scope-noop="static_callback"'
+            . '></p>';
+
+        $secondCloneHtml = '<p '
+            . 'class="second_clone_class" '
+            . 'test-instance-scope-noop-inline="inline" '
+            . 'test-instance-noop-attribute="static_callback" '
+            . 'test-closure-static-scope-noop="static_callback" '
+            . 'test-closure-instance-scope-noop="static_callback"'
+            . '></p>';
+
+        $this->assertSame($originalHtml, $original->render());
+        $this->assertSame($firstCloneHtml, $firstClone->render());
+        $this->assertSame($secondCloneHtml, $secondClone->render());
+    }
+
+    public function testElementCallbacksCloning(): void
+    {
+        $element = new CloningDummyElement();
```

Comment:
```suggestion
        $element = new CloningDummyElement();
        $element->getAttributes();
```

Otherwise the test is useless. (Still runs through on master)